File: backend/src/main/java/de/snickit/rezeptebackend/entities/Ingredient.java
package de.snickit.rezeptebackend.entities;

import javax.persistence.*;
import java.util.Collection;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "ingredient", schema = "public")
public class Ingredient {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long ingredientId;

    @OneToMany
    private Collection<RecipeStepIngredients> recipeStepIngredients;

    public Long getIngredientId() {
        return ingredientId;
    }

    public void setIngredientId(Long ingredientId) {
        this.ingredientId = ingredientId;
    }
}

File: backend/src/main/kotlin/de/snickit/recipe/backend/persistence/entities/Ingredient.kt
package de.snickit.recipe.backend.persistence.entities

import com.fasterxml.jackson.annotation.JsonBackReference
import javax.persistence.*

@Entity
@SequenceGenerator(name = "ingredientSeq",
        sequenceName = "INGREDIENT_SEQUENCE",
        allocationSize = 1,
        initialValue = 7000)
@Table
data class Ingredient(
        @Column
        var name: String? = null
) {
    @Id
    @GeneratedValue(generator = "ingredientSeq")
    var ingredientId: Long? = null
}
